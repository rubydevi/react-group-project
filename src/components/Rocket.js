import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../Redux/Rockets/rocketsSlice';
import '../styles/Rocket.css';

const Rocket = ({ rocket }) => {
  const dispatch = useDispatch();

  const handleReserveRocket = () => {
    dispatch(reserveRocket(rocket.id));
  };

  return (
    <li className="rocket" key={rocket.id}>
      {rocket.flickr_images.length > 0 && (
      <img className="rocket-image" src={rocket.flickr_images[0]} alt={`Rocket ${rocket.name}`} />
      )}
      <div className="rocket-details">
        <h3 className="rocket-title">{rocket.name}</h3>
        {/* Show badge when reserved */}
        {rocket.reserved && <span className="reserved-badge">Reserved</span>}
        <p className="rocket-description">
          {rocket.description}
        </p>
        <button className="reserve-button" type="submit" onClick={handleReserveRocket}>
          Reserve Rocket
        </button>
      </div>
    </li>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
    reserved: PropTypes.bool.isRequired,
  }).isRequired,
};

export default Rocket;
