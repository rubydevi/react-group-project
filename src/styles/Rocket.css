.rocket {
  display: flex;
  column-gap: 20px;
}

.rocket-image {
  width: 27%;
}

.rocket-description {
  font-size: 1.2em;
  font-weight: 500;
}

button {  
  padding: 8px 10px;
  border-radius: 5px;
  border: none;
  outline: none;
}

.reserved-badge {
  float: left;
  padding: 1px 5px;
  background-color: #18a2b8;
  color: #f7fcfc;
  border-radius: 10%;
  margin-right: 5px;
}

.reserve-button {
  background-color: #007bff;
  color: #fff;
}

.reserve-button:active {
  background-color: #0069da;
}

.reserve-button:hover {
  background-color: #0071eb;
}

.cancel-button {
  border: 1px solid #6f777f;
  color: #6d757d;
  background-color: #fff;
}
