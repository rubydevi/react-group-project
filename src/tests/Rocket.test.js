import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Rocket from '../components/Rocket';

// Create a mock store with the state you want to use in the test
const mockStore = configureStore([]);

test('Rocket component renders correctly', () => {
  const mockRocket = {
    id: '5e9d0d95eda69955f709d1eb',
    name: 'Falcon 1',
    description: 'Description 1',
    flickr_images: ['https://imgur.com/DaCfMsj.jpg'],
    reserved: false,
  };

  const store = mockStore({});

  const tree = renderer.create(
    <Provider store={store}>
      <Rocket rocket={mockRocket} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
