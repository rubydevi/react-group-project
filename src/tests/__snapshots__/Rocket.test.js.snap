// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component renders correctly 1`] = `
<li
  className="rocket"
>
  <img
    alt="Rocket Falcon 1"
    className="rocket-image"
    src="https://imgur.com/DaCfMsj.jpg"
  />
  <div
    className="rocket-details"
  >
    <h3
      className="rocket-title"
    >
      Falcon 1
    </h3>
    <div>
      <p
        className="rocket-description"
      >
        Description 1
      </p>
    </div>
    <button
      className="reserve-button"
      onClick={[Function]}
      type="submit"
    >
      Reserve Rocket
    </button>
  </div>
</li>
`;
